ir/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -488,7 +493,7 @@ object Trees {
     val tpe = AnyType
   }
 
-  case class JSPrivateSelect(qualifier: Tree, item: Ident)(
+  case class JSPrivateSelect(qualifier: Tree, cls: ClassRef, field: Ident)(
```

Comment:
The typing rules are significantly different:
* `Select` requires the `qualifier` to be of a subtype of `cls`, whereas `JSPrivateSelect` accepts anything
* The `tpe` of a `Select` is the `ftpe` of the `FieldDef`, but the `tpe` of a `JSPrivateSelect` is always `any`

and the encoding in JS could and will be significantly different as well (to deal with #2680 and #2748).

We don't have the distinction for static methods because the typing rules and the encoding in JS are exactly the same.

Reply:
> The typing rules are significantly different

OK. But is/should `JSPrivateSelect` ever be used such that the stricter rules are not observed? For example, why should it be OK to `JSPrivateSelect` on anything where the same is not allowed for private methods?

> encoding in JS could and will be significantly different as well

But this is entirely determined by what kind of class the field is defined on, right?

All in all, I just find the non-duality between `FieldDef` and `Select` / `JSPrivateSelect` a bit strange to understand. I'd find a model where `FieldDef`s can be `Select'ed and the encoding of this just happens to be different for JS / non JS classes easier to understand.